### ISSUE TITLE
Microfiber options (especially removeUnusedTypes) need to be exposed

### ISSUE THREAD

**Issue by felixvd** [anvilco/spectaql#437]
I just spent way too long hunting down a parameter that caused many types in my schema to be hidden, and which was not accessible via `config.yaml`. I had to change [these lines](https://github.com/anvilco/spectaql/blob/787524916133a9921858dfa4452e12edfb8d0d2f/src/themes/default/data/index.js#L6--8) in `src/themes/default/data/index.js` to display all the types:

```ts
  let opts
  opts = {
    removeUnusedTypes: false
  }
  const introspectionManipulator = new IntrospectionManipulator(
    introspectionResponse, opts
  )
```

This option should be in the top-level `config.yaml` instead of inaccessible.

This was especially frustrating because the documentation instructs the user to set `"documented": true` to unhide types, but that does not affect the ones dropped by this hidden setting.

The setting also mistakenly considers types unused if they implement an interface which is referenced, but are not referenced directly. 

Thanks for sharing the package otherwise. It has been very helpful.

**Comment by newhouse** [comment]
Hi @felixvd and thanks for using SpectaQL. Sorry about your issue.

Hmm, yes, the options you need should actually be available in the "hide" options, like [this one](https://github.com/anvilco/spectaql/blob/main/config-example.yml#L296).

However, it does look like I'm not passing them to that location like I should be so that they are used in that instance of microfiber.

Can you tell me what was happening? You had some Types that were not used anywhere that were not displayed, but you wanted them to be?

**Comment by felixvd** [comment]
Yes, that's right. Concretely, my schema is referencing an interface, but the types implementing the interface are dropped:

```graphql
type Query {
  ListStats(
    statType: String
  ): ListStatReturn
}

type ListStatReturn {
  rets: [StatEntry!]!
}

interface StatEntry {
  metainfo: String
}

type Stat1 implements StatEntry {
  metainfo: String
  value: Int
}

type Stat2 implements StatEntry {
  metainfo: String
  value: Int
}
```
(Stat1 and Stat2 do not appear in the doc)

This minimal example SDL reproduces the issue as well:

```graphql
type Query {
  DummyQuery(
        nofield: String
  ): MyTestType1
}

type MyTestType1 {
  MyTestField: String
}

type MyTestType2 {
  MyTestField1: Int
}
```

`MyTestType2` is removed from the doc regardless of the `config`'s content. This is the first problem. I came across another after this.

Setting `removeUnusedFields=false` seems to cause the `underlyingTypeDefinition` not to be loaded correctly for types that are not referenced by a query (and are not `String` or `Boolean`). You can reproduce it with this SDL and the change described in the original post:

```graphql
type Query {
  MyQuery(
    ret: String
  ): String
}

type OtherType {
  s: String
  b: Boolean
  f: Float
  i: Int
  o: OtherType2
}

type OtherType2 {
  s: String
}
```

I can't post my actual SDL file publicly, but I am happy to send it privately. It may clarify what I am trying to do and avoid an x-y problem.

**Comment by felixvd** [comment]
I would have liked to read through microfiber's [`cleanSchema`](https://github.com/anvilco/graphql-introspection-tools/blob/main/src/microfiber.js#L509) to see if it can be changed such that types which implement a referenced interface will not be removed. But it does not look like I will have the time. I hope the change is easy and you or someone else (or me if I can find time in the future) can fix it.

For completeness, I wanted to add that I needed to add these kinds of examples (with fragments) to our generated doc, and that it would have been very useful to have them auto-generated as well:

```json graphql
query {
     DummyQuery (
        ret: "stat1"
     ) {
      ... on Stat1 {
        value
      }
      ... on Stat2 {
        othervalue
      }
```

Hope this use case can serve as a reference. Thanks again for the tool!

**Comment by newhouse** [comment]
Ah yes, you're only referencing the _interface_ type and not the implementations...this is interesting.

Yeah, I think now that what I should do primarily is to update `microfiber` to consider types that _implement_ an interface that is used as something to keep around. Let me check that out.

**Comment by newhouse** [comment]
This is fixed/provided for in `main` and v`1.5.2`.

**Comment by felixvd** [comment]
Excellent, thank you :)
